Decompiler fails when there's no java in path

### Describe the bug

Decompiler fails when there's no java in path:

```
$ JAVA_HOME=/opt/graalvm-ce-java17-22.3.0 /opt/maven/bin/mvn package -DskipTests -Dquarkus.package.quiltflower.enabled
...
[ERROR] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Failed to launch decompiler.
java.io.IOException: Cannot run program "java": error=2, No such file or directory
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1143)
    at java.lang.ProcessBuilder.start (ProcessBuilder.java:1073)
    at io.quarkus.deployment.pkg.steps.JarResultBuildStep$Decompiler$QuiltflowerDecompiler.decompile (JarResultBuildStep.java:1628)
    at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildThinJar (JarResultBuildStep.java:642)
    at io.quarkus.deployment.pkg.steps.JarResultBuildStep.buildRunnerJar (JarResultBuildStep.java:222)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at io.quarkus.deployment.ExtensionLoader$3.execute (ExtensionLoader.java:909)
    at io.quarkus.builder.BuildContext.run (BuildContext.java:282)
    at org.jboss.threads.ContextHandler$1.runWith (ContextHandler.java:18)
    at org.jboss.threads.EnhancedQueueExecutor$Task.run (EnhancedQueueExecutor.java:2513)
    at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run (EnhancedQueueExecutor.java:1538)
    at java.lang.Thread.run (Thread.java:833)
    at org.jboss.threads.JBossThread.run (JBossThread.java:501)
```

The decompiler code should be taking into account `JAVA_HOME`.

### Expected behavior

Decompiler should work when no `java` is in path but `JAVA_HOME` is provided.

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

9fa600951e9

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
@@ -88,6 +88,7 @@ import io.quarkus.maven.dependency.GACT;
 import io.quarkus.maven.dependency.ResolvedDependency;
 import io.quarkus.paths.PathVisit;
 import io.quarkus.paths.PathVisitor;
+import io.quarkus.utilities.JavaBinFinder;
 
 /**
  * This build step builds both the thin jars and uber jars.
@@ -1612,7 +1613,7 @@ public class JarResultBuildStep {
                     String fileName = jarToDecompile.getFileName().toString().substring(0, dotIndex);
                     ProcessBuilder processBuilder = new ProcessBuilder(
                             Arrays.asList(
-                                    "java",
+                                    JavaBinFinder.findBin(),
                                     "-jar",
                                     decompilerJar.toAbsolutePath().toString(),
                                     "-rsy=0", // synthetic methods

